src/types/index.ts

```diff
@@ -1,42 +1,94 @@
+/* eslint-disable no-unused-vars */
 /**
  * Some options from `vite.config.[ts|js]`
  * @since 0.2.0
  */
 export interface BannerPluginOptions {
   /**
    * The comment content of the banner
+   *
+   * @since 0.6.0
+   *
+   * callback function available since 0.6.0
+   * @example <caption>content Callback(since 0.6.0)</caption>
+   * ```ts
+   * content: (fileName: string) => fileName.endsWith('.js') ? 'this message will inject into js file' : ''
+   * // inject into js file, but not inject into css file
+   * // You can also continue to write other flows.
+   * ```
+   * @param fileName - The name of the file
+   * @returns {string | contentCallback} What want to inject into the file. More details see {@link contentCallback}
    */
-  content: string
+  content: string | contentCallback
 
   /**
    * The output directory from the configuration of Vite.js
-   * @default `dist`
+   * @default 'dist'
    */
   outDir?: string
 
   /**
    * Whether to print error messages to the console
    * @since 0.4.0
-   * @default `false`
+   * @default false
    */
   debug?: boolean
 
   /**
    * By default, the validity of the content will be verified.
+   *
    * If set to `false`, no verification will be performed.
    * @see https://github.com/chengpeiquan/vite-plugin-banner/issues/13
    * @since 0.5.0
-   * @default `true`
+   * @default true
    */
   verify?: boolean
+
+  /**
+   * Exclude specified files.
+   *
+   * support `strings`, `RegExp`, and `function`.
+   * @example <caption>Exclude css files</caption>
+   * exclude: '.css'
+   * // or
+   * exclude: /\.css/
+   * // or
+   * exclude: (file) => file.endsWith('.css')
+   * @since 0.6.0
+   * @default undefined
+   * @param fileName - The file name
+   * @returns {boolean} Whether to exclude the file. See more: {@link excludeCallback}
+   */
+  exclude?: string | RegExp | excludeCallback
```

Comment:
I was thinking it might be better to use `include` to include the file extension:

- When CSS files need to be processed, there may only be a case of `.css` extension
- When JS files need to be processed, various extensions such as `.js` / `.cjs` / `.mjs` may be included

Use exclude in unknown situations, for example, when processing CSS, only exclude `.js` and forget `.cjs` and `.mjs`, there will be unexpected situations, I suggest changing to include and using  format of `string[]` instead of `string`

Reply:
In fact, I think that because the callback function is already allowed to determine the filename directly, it allows the user to determine the type of file to be processed (e.g. html or other files). So whether it is still necessary to restrict the css and js file types?

The `exclude` here is also simply because It will inject the same comments into the css when injecting js causing an error. I wasn't sure if this was a good way to go, but it did solve the problem I had at the time.

As with the commits that follow, I found this problematic after using it for some time (using `exclude` causes confusion), so I added function callbacks, and I think I can just remove the exclude argument, as the callback function with the function name contains this functionality entirely, don't you think?